test/jdk/java/net/CookieStoreTest.java

```diff
@@ -0,0 +1,112 @@
+        final List<Arguments> params = new ArrayList<>();
+        // empty CookieStore
+        params.add(Arguments.of(new CookieManager().getCookieStore(), true));
+
+        final CookieStore cookieStore = new CookieManager().getCookieStore();
+        cookieStore.add(COOKIE_TEST_URI, new HttpCookie("foo", "bar"));
+        // non-empty CookieStore
+        params.add(Arguments.of(cookieStore, false));
+
+        return params;
+    }
+
+    /*
+     * Verify that the List returned by CookieStore.getURIs() is immutable.
+     */
+    @ParameterizedTest
+    @MethodSource("cookieStores")
+    void testImmutableGetURIs(final CookieStore cookieStore, final boolean expectEmpty) {
+        final List<URI> uris = cookieStore.getURIs();
+        assertNotNull(uris, "CookieStore.getURIs() returned null");
+        assertEquals(expectEmpty, uris.isEmpty(), "CookieStore.getURIs() returned: " + uris);
+
+        // the returned list is expected to be immutable, so the attempt to add or remove
+        // an element must fail
+        assertThrows(UnsupportedOperationException.class, () -> uris.add(COOKIE_TEST_URI));
+        assertThrows(UnsupportedOperationException.class, () -> uris.remove(COOKIE_TEST_URI));
+    }
+
+    /*
+     * Verify that the List returned by CookieStore.getCookies() is immutable.
+     */
+    @ParameterizedTest
+    @MethodSource("cookieStores")
+    void testImmutableGetCookies(final CookieStore cookieStore, final boolean expectEmpty) {
+        final List<HttpCookie> cookies = cookieStore.getCookies();
+        assertNotNull(cookies, "CookieStore.getCookies() returned null");
+        assertEquals(expectEmpty, cookies.isEmpty(), "CookieStore.getCookies() returned: " + cookies);
+
+        final HttpCookie cookie = new HttpCookie("hello", "world");
+        // the returned list is expected to be immutable, so the attempt to add or remove
+        // an element must fail
+        assertThrows(UnsupportedOperationException.class, () -> cookies.add(cookie));
+        assertThrows(UnsupportedOperationException.class, () -> cookies.remove(cookie));
+    }
+
+    /*
+     * Verify that the List returned by CookieStore.get(URI) is immutable.
+     */
+    @ParameterizedTest
+    @MethodSource("cookieStores")
+    void testImmutableGetCookiesForURI(final CookieStore cookieStore, final boolean expectEmpty) {
+        final List<HttpCookie> cookies = cookieStore.get(COOKIE_TEST_URI);
+        assertNotNull(cookies, "CookieStore.get(URI) returned null");
+        assertEquals(expectEmpty, cookies.isEmpty(), "CookieStore.get(URI) returned: " + cookies);
+
+        final HttpCookie cookie = new HttpCookie("hello", "world");
+        // the returned list is expected to be immutable, so the attempt to add or remove
+        // an element must fail
+        assertThrows(UnsupportedOperationException.class, () -> cookies.add(cookie));
+        assertThrows(UnsupportedOperationException.class, () -> cookies.remove(cookie));
```

Comment:
Nit: You can consider factoring this out to a `assertCookieStoreIsImmutable(CookieStore)` method.

Reply:
Done. I've moved couple of those assertions to a separate method to be called from each of these test methods.